src/org.hdfgroup.hdfview/hdf/view/dialog/UserOptionsHDFPage.java

```diff
@@ -373,6 +436,66 @@ public void widgetSelected(SelectionEvent e) {
         checkNativeOrder.setText("Native");
         checkNativeOrder.setLayoutData(new GridData(SWT.CENTER, SWT.FILL, true, false));
 
+        org.eclipse.swt.widgets.Group pluginDirectoryGroup = new org.eclipse.swt.widgets.Group(composite, SWT.NONE);
+        pluginDirectoryGroup.setLayout(new GridLayout(3, false));
+        pluginDirectoryGroup.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, false));
+        pluginDirectoryGroup.setFont(curFont);
+        pluginDirectoryGroup.setText("Default Plugin Directory (*/plugin)");
+
+        checkCurrentUserDir = new Button(pluginDirectoryGroup, SWT.CHECK);
+        checkCurrentUserDir.setFont(curFont);
+        checkCurrentUserDir.setText("\"User Work\" or");
+        checkCurrentUserDir.setLayoutData(new GridData(SWT.FILL, SWT.FILL, false, false));
+        checkCurrentUserDir.addSelectionListener(new SelectionAdapter() {
+            @Override
+            public void widgetSelected(SelectionEvent e) {
+                boolean isCheckCurrentUserDirSelected = checkCurrentUserDir.getSelection();
+                if (isCheckCurrentUserDirSelected)
+                    checkUserHomeDir.setSelection(false);
+                pluginField.setEnabled(!isCheckCurrentUserDirSelected);
+                currentDirButton.setEnabled(!isCheckCurrentUserDirSelected);
+            }
+        });
+
+        checkUserHomeDir = new Button(pluginDirectoryGroup, SWT.CHECK);
+        checkUserHomeDir.setFont(curFont);
+        checkUserHomeDir.setText("\"User Home\" or");
+        checkUserHomeDir.setLayoutData(new GridData(SWT.FILL, SWT.FILL, false, false));
+        checkUserHomeDir.addSelectionListener(new SelectionAdapter() {
+            @Override
+            public void widgetSelected(SelectionEvent e) {
+                boolean isCheckUserHomeDirSelected = checkUserHomeDir.getSelection();
+                if (isCheckUserHomeDirSelected)
+                    checkCurrentUserDir.setSelection(false);
+                pluginField.setEnabled(!isCheckUserHomeDirSelected);
+                currentDirButton.setEnabled(!isCheckUserHomeDirSelected);
+            }
+        });
+
+        pluginField = new Text(pluginDirectoryGroup, SWT.SINGLE | SWT.BORDER);
+        pluginField.setFont(curFont);
+        pluginField.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, false));
+
+        currentDirButton = new Button(pluginDirectoryGroup, SWT.PUSH);
+        currentDirButton.setFont(curFont);
+        currentDirButton.setText("Browse...");
+        currentDirButton.setLayoutData(new GridData(SWT.FILL, SWT.FILL, false, false));
+        currentDirButton.addSelectionListener(new SelectionAdapter() {
+            @Override
+            public void widgetSelected(SelectionEvent e) {
```

Comment:
This should probably just let users select the plugin dir rather than a base directory. I think that will be confusing to a lot of people

Reply:
I have been thinking more about this, since this proof of implementation.  It needs to be a list of paths and editable. At which time I think I will just drop that extra "/plugin" and require the real value.